paymentintent/client.go

```diff
@@ -0,0 +1,131 @@
+	path := stripe.FormatURLPath("/payment_intents/%s", id)
+	intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodGet, path, c.Key, params, intent)
+	return intent, err
+}
+
+// Update updates a payment intent.
+func Update(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Update(id, params)
+}
+
+// Update updates a payment intent.
+func (c Client) Update(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s", id)
+	intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, intent)
+	return intent, err
+}
+
+// Cancel cancels a payment intent.
+func Cancel(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Cancel(id, params)
+}
+
+// Cancel cancels a payment intent.
+func (c Client) Cancel(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/cancel", id)
+	intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, intent)
+	return intent, err
+}
+
+// Capture captures a payment intent.
+func Capture(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Capture(id, params)
+}
+
+// Capture captures a payment intent.
+func (c Client) Capture(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/capture", id)
+	intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, intent)
+	return intent, err
+}
+
+// Confirm confirms a payment intent.
+func Confirm(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	return getC().Confirm(id, params)
+}
+
+// Confirm confirms a payment intent.
+func (c Client) Confirm(id string, params *stripe.PaymentIntentParams) (*stripe.PaymentIntent, error) {
+	path := stripe.FormatURLPath("/payment_intents/%s/confirm", id)
+	intent := &stripe.PaymentIntent{}
+	err := c.B.Call(http.MethodPost, path, c.Key, params, intent)
+	return intent, err
+}
+
+// List returns a list of payment intents.
+func List(params *stripe.PaymentIntentListParams) *Iter {
+	return getC().List(params)
+}
+
+// List returns a list of payment intents.
+func (c Client) List(listParams *stripe.PaymentIntentListParams) *Iter {
+	return &Iter{stripe.GetIter(listParams, func(p *stripe.Params, b *form.Values) ([]interface{}, stripe.ListMeta, error) {
+		list := &stripe.PaymentIntentList{}
+		err := c.B.CallRaw(http.MethodGet, "/payment_intents", c.Key, b, p, list)
+
+		ret := make([]interface{}, len(list.Data))
+		for i, v := range list.Data {
+			ret[i] = v
+		}
+
+		return ret, list.ListMeta, err
+	})}
+}
+
+// Iter is an iterator for lists of PaymentIntents.
+// The embedded Iter carries methods with it;
+// see its documentation for details.
+type Iter struct {
```

Comment:
I think we ended up updating all these `Iter` comments in your other Lint PRs. Can we change this one too? (And maybe the comment on `PaymentIntent()` directly below too.

Reply:
Nice catch, fixed other comments too